storage2/src/storage.rs

```diff
@@ -59,15 +57,16 @@ impl Storage {
     /// `Storage`, or `None` if the tree is empty.
     pub async fn latest_version(&self) -> Result<Option<jmt::Version>> {
         // TODO: do better
-        Ok(latest_version(self.0.db).unwrap())
+        latest_version(&self.0.db)
     }
 
     /// Returns a new [`State`] on top of the latest version of the tree.
-    pub async fn state(&self) -> State {
+    pub fn state(&self) -> State {
         State::new(self.0.latest_snapshot.read().clone())
     }
 
-    pub async fn apply(&'static mut self, state: State) -> Result<()> {
+    pub async fn apply(&self, state: State) -> Result<()> {
```

Comment:
This takes `&self`, not `&mut self`, because it's handling synchronization internally, so it doesn't need an exclusive reference (cf [this distinction](https://docs.rs/dtolnay/latest/dtolnay/macro._02__reference_types.html))